mantidimaging/gui/windows/main/presenter.py

```diff
@@ -168,12 +168,22 @@ def _on_stack_load_done(self, task: 'TaskWorkerThread') -> None:
         else:
             self._handle_task_error(self.LOAD_ERROR_STRING, task)
 
+    def _launch_window_if_cli_flag_set(self) -> None:
+        """
+        Launches windows that require loaded data if the CLI flags are set.
+        """
+        if self.view.args.operation():
+            self.show_operation(self.view.args.operation())
+        if self.view.args.recon():
+            self.view.show_recon_window()
+
     def _on_dataset_load_done(self, task: 'TaskWorkerThread') -> None:
 
         if task.was_successful():
             self._add_strict_dataset_to_view(task.result)
             self.view.model_changed.emit()
             task.result = None
+            self._launch_window_if_cli_flag_set()
```

Comment:
Will this cause the operation or recon window to reopen if the user loads another dataset?

Maybe `load_stacks_from_folder()` could create a function like
```python
    def on_load_and_finish_cli_processing(task):
        self._on_dataset_load_done(task)
        self._finish_processing_cli_args()
```
And pass that to `start_async_task_view()`

Reply:
Nice catch. Doesn't seem to reopen as the widow is already open, but it does change the focus to place the recon window over the main window again as the active window so current implementation will need altered slighty to avoid this.